client/interaction/Scanner.py

```diff
@@ -1,30 +1,92 @@
 from threading import Thread
+from time import sleep
+from client.interaction.Door import Door
 
 from client.interaction.InteractionMode import InteractionMode
+from client.interaction.Placement import Placement
+from client.networking.Host import Host
+from client.persistance.Authorization import Authorization
+from client.persistance.EventType import EventType
+from client.persistance.DatabaseManager import DatabaseManager
+from client.persistance.types.User import User
 from rfid_library import SimpleMFRC522
 from logging import info, debug
 
 
 class Scanner(Thread):
-    def __init__(self, device_id):
+    def __init__(self, device_id, host: Host, door: Door, database_manager: DatabaseManager):
         super().__init__()
         self.device_id = device_id
+        self.database_manager = database_manager
         debug(f"Initializing scanner {self.device_id}")
         self.board = SimpleMFRC522(bus=0, device=device_id)
         debug(f"Scanner {self.device_id} initialized")
         self.mode = InteractionMode.NO_ACTION
+        self.placement = Placement.ENTRANCE if device_id == 0 else Placement.EXIT
+        self.host = host
+        self.door = door
+        self.is_modify_mode = False
 
     def run(self):
         info(f"Scanner {self.device_id} started")
         while True:
-            id, text = self.board.read()
-            info(id)
-            info(text)
-            info(self.mode)
+            tag_id = self.board.read()[0]
+            info(tag_id)
             if self.mode == InteractionMode.READ:
+                user = self.database_manager.get_user(tag_id)
+                is_authorized = Authorization.UNAUTHORIZED.value if user is None else int(
+                    user.is_authorized)
```

Comment:
Wielokrotnie powtarza się porównywanie is_authorized po intach. Przydało by się żeby is_authorized był enumem. Możliwe że wystarczy podać do konstruktora Authorization tego inta, a jeśli sie nie to dodaj metodke matchującą inta na enum

Reply:
w sumie to is_authorized zwraca enuma, nwm czy da się od buta porównywać enumy w pythongu, ale jak tak to chyba wystarczy usunąć .value i castowanie na inta